src/reanimated2/hook/useAnimatedRef.ts

```diff
@@ -19,15 +23,19 @@ const getTagValueFunction = global._IS_FABRIC
   ? getShareableShadowNodeFromComponent
   : getTag;
 
-export function useAnimatedRef<T extends Component>(): RefObjectFunction<T> {
+export function useAnimatedRef<T extends ComponentRef>(): RefObjectFunction<T> {
   const tag = useSharedValue<number | ShadowNodeWrapper | null>(-1);
   const ref = useRef<RefObjectFunction<T>>();
 
   if (!ref.current) {
     const fun: RefObjectFunction<T> = <RefObjectFunction<T>>((component) => {
       // enters when ref is set by attaching to a component
       if (component) {
-        tag.value = getTagValueFunction(component);
+        tag.value = getTagValueFunction(
+          component.getScrollableNode
+            ? component.getScrollableNode()
+            : component
```

Comment:
Maybe, we could add some comment why we use `.getScrollableNode()` or just add information that it comes from `FlashList`?

Reply:
Hopefully `git blame` should find this PR which links to the original issue but yeah you're right, adding a comment here would be nice 😄 Also, there's a similar place in `createAnimatedComponent.tsx` which does exactly the same thing and I'd like to move it to a new utils function, wdyt?

Reply:
The idea of a new function sound great ☺️